substrait/common/NumberUtils.h

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#pragma once
+
+#include <string>
+
+namespace io::substrait::common {
+
+class NumberUtils {
+ public:
+  static bool isInteger(std::string_view s) {
+    return isNonNegativeInteger(s) || isNegativeInteger(s);
+  }
+
+  static bool isNonNegativeInteger(std::string_view s) {
+    return !s.empty() && std::all_of(s.begin(), s.end(), ::isdigit);
```

Comment:
```suggestion
    return !s.empty() && std::all_of(s.begin(), s.end(), [](unsigned char d) {
      return std::isdigit(d);
    });
```
This is needed according to the note in https://en.cppreference.com/w/cpp/string/byte/isdigit